### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by LouisDISPA**
Hello, thank you for this amazing work. It works on my machine with only a small issue. (the color profile for thumbnails)
I quickly updated the dependencies to see if this would fix it but sadly it did not.

Here is a PR with the updates, you are free to do whatever you want with it.
If you have any request of modification feel free to ask. I have some spare time.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,9 +9,11 @@ A JPEG XL (*.jxl) WIC decoder to render thumbnails on Windows File Explorer or v
 1. Move to your download directory
 1. `regsvr32 jxl_winthumb.dll`, or to uninstall, `regsvr32 /u jxl_winthumb.dll`.
 
+You might need to restart windows for the extention to work.
```
Comment (saschanaz) [inline README.md:12]: Restarting `explorer.exe` should be enough.

Comment (saschanaz) [reply README.md:12]:
```suggestion
You might need to restart `explorer.exe` for the component to work.
```

("Component" because C of both WIC and COM is Component.)

**Review by saschanaz — COMMENTED** [review]
Wow, thank you for the chore! Yeah, I think fixing the issue needs some more investigation (I think the issue might just be about emitting the 128bit float pixel because I guess that's a rare use case. I wonder the new jxl-oxide supports lower precision number types?

Some nits:

File: tests/wic.rs [hunk 2]
```diff
@@ -5,7 +5,7 @@ use windows::Win32::UI::Shell::SHCreateMemStream;
 
 #[test]
 fn basic() {
-    unsafe { CoInitialize(None) }.expect("CoInitialize");
+    unsafe { CoInitialize(None) }.unwrap();
```
Comment (saschanaz) [inline tests/wic.rs:8]: Why this change? 🤔

Comment (LouisDISPA) [reply tests/wic.rs:8]: CoInitialize now return a custom type that only have `unwrap` and not `expect`. But I found a method to change it to a standard `Result` type.

**Comment by LouisDISPA** [comment]
Welcome, I fixed the nits!  
I also added the cmd to restart `explorer.exe`.

> I think fixing the issue needs some more investigation (I think the issue might just be about emitting the 128bit float pixel because I guess that's a rare use case. I wonder the new jxl-oxide supports lower precision number types?

Good to know, I don't have a lot of experience on image processing but I might try to have a look.

**Comment by LouisDISPA** [comment]
I pushed a last commit that remove some unwrap by using pattern matching. The logic is the same but if you prefer the old code I can revert it.

File: Cargo.toml [hunk 3]
```diff
@@ -10,13 +10,13 @@ crate-type = ["rlib", "cdylib"]
...
 
 [dependencies]
-simple-logging = "2.0.2"
-log = "0.4.17"
-winreg = "0.10.1"
-jxl-oxide = "0.5.1"
+simple-logging = "2.0"
+log = "0.4"
+winreg = "0.52.0"
+jxl-oxide = "0.7"
```
Comment (goodusername123) [inline Cargo.toml:16]: Version [0.7.1](https://github.com/tirr-c/jxl-oxide/releases/tag/0.7.1) of jxl-oxide released a few hours ago.

**Comment by saschanaz** [comment]
> I pushed a last commit that remove some unwrap by using pattern matching. The logic is the same but if you prefer the old code I can revert it.

I do like them but diffing is now harder, can we do that in a separate PR?

File: Cargo.toml [hunk 4]
```diff
@@ -10,13 +10,13 @@ crate-type = ["rlib", "cdylib"]
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-simple-logging = "2.0.2"
-log = "0.4.17"
-winreg = "0.10.1"
-jxl-oxide = "0.5.1"
+simple-logging = "2.0"
+log = "0.4"
```
Comment (saschanaz) [inline Cargo.toml:14]: Was this done by some tool or by hand? Not sure the point to remove the patch versions.

File: README.md [hunk 5]
```diff
@@ -9,9 +9,12 @@ A JPEG XL (*.jxl) WIC decoder to render thumbnails on Windows File Explorer or v
 1. Move to your download directory
 1. `regsvr32 jxl_winthumb.dll`, or to uninstall, `regsvr32 /u jxl_winthumb.dll`.
 
+You might need to restart `explorer.exe` for the component to work.  
+You can use this command: `taskkill /f /im explorer.exe & start explorer.exe`
```
Comment (saschanaz) [inline README.md:13]:
```suggestion
You can use this command: `taskkill /f /im explorer.exe && start explorer.exe`
```

`&` does different thing in pwsh, while it seems both work in cmd.

File: src/properties.rs [hunk 6]
```diff
@@ -1,21 +1,21 @@
 use std::io::BufReader;
 
+use crate::winstream::WinStream;
```
Comment (saschanaz) [inline src/properties.rs:3]: I kinda tried to put all internal imports after external imports although there was an exception. Meaning, I'd be happier to keep WinStream after windows imports.

File: src/dll.rs [hunk 7]
```diff
@@ -113,7 +113,7 @@ pub extern "stdcall" fn DllMain(
 pub unsafe extern "system" fn DllGetClassObject(
     rclsid: *const GUID,
     riid: *const GUID,
-    pout: *mut *const core::ffi::c_void,
+    pout: *mut *mut core::ffi::c_void,
```
Comment (saschanaz) [inline src/dll.rs:116]: Oh, unknown.query now accepts mut mut. Seems it also accepts `*const GUID` instead of a reference now, can you remove `&*` there while we are at it?

**Review by saschanaz — COMMENTED** [review]
Looking good! A few more nits:

File: Cargo.toml (hunk 4 above)
Comment (LouisDISPA) [reply Cargo.toml:14]: This was done by hand. Since the logging crates are stable I removed the patch versions. Maybe jxl-oxide should have it just to be sure. As you wish!

File: src/properties.rs (hunk 6 above)
Comment (LouisDISPA) [reply src/properties.rs:3]: Sorry it's the cargo fmt command that moved everything.

File: src/dll.rs (hunk 7 above)
Comment (LouisDISPA) [reply src/dll.rs:116]: Good catch!

**Comment by LouisDISPA** [comment]
> Looking good! A few more nits:
Done!

File: README.md [hunk 8]
```diff
@@ -9,9 +9,12 @@ A JPEG XL (*.jxl) WIC decoder to render thumbnails on Windows File Explorer or v
 1. Move to your download directory
 1. `regsvr32 jxl_winthumb.dll`, or to uninstall, `regsvr32 /u jxl_winthumb.dll`.
 
+You might need to restart `explorer.exe` for the component to work.  
+You can use this command: `taskkill /f /im explorer.exe && start explorer.exe`
```
Comment (saschanaz) [inline README.md:13]:
```suggestion
You might need to restart `explorer.exe` or any programs that use the dll before updating it. Get the list of such programs using `tasklist /m jxl_winthumb.dll` and kill them e.g. with `taskkill /f /im explorer.exe && start explorer.exe`.
```

**Comment by saschanaz** [comment]
Thank you so much! So helpful 👍🙏
